src/main/java/com/celements/auth/user/listener/EnsureConsistentUserStateListener.java

```diff
@@ -0,0 +1,104 @@
+package com.celements.auth.user.listener;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.springframework.stereotype.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.observation.event.Event;
+
+import com.celements.auth.user.UserPageType;
+import com.celements.common.observation.listener.AbstractLocalEventListener;
+import com.celements.model.object.xwiki.XWikiObjectEditor;
+import com.celements.model.util.ModelUtils;
+import com.celements.observation.save.SaveEventOperation;
+import com.celements.observation.save.object.ObjectEvent;
+import com.celements.pagetype.classes.PageTypeClass;
+import com.celements.rights.access.EAccessLevel;
+import com.celements.web.classes.oldcore.XWikiRightsClass;
+import com.celements.web.classes.oldcore.XWikiUsersClass;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.user.api.XWikiUser;
+
+@Component
```

Comment:
this class implements the `ComponentRole` `EventListener` thus we need to set a name:
```suggestion
@Component(EnsureConsistentUserStateListener.NAME)
```
if no name is set it defaults to the name `default` which could cause issues due to duplicate namings. if it weren't a  `ComponentRole` it'd default to the class FQN as the name. in these cases you don't need to set an explicit name.